platform-api/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/LinksInjector.java

```diff
@@ -0,0 +1,117 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.user.server;
+
+import org.eclipse.che.api.core.rest.ServiceContext;
+import org.eclipse.che.api.core.rest.shared.dto.Link;
+import org.eclipse.che.api.core.util.LinksHelper;
+import org.eclipse.che.api.user.shared.dto.UserDescriptor;
+import org.eclipse.che.commons.env.EnvironmentContext;
+import org.eclipse.che.commons.user.User;
+
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.core.UriBuilder;
+import java.util.LinkedList;
+import java.util.List;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_FORM_URLENCODED;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_GET_CURRENT_USER;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_GET_CURRENT_USER_PROFILE;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_GET_USER_BY_EMAIL;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_GET_USER_BY_ID;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_GET_USER_PROFILE_BY_ID;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_REMOVE_USER_BY_ID;
+import static org.eclipse.che.api.user.server.Constants.LINK_REL_UPDATE_PASSWORD;
+
+/**
+ * Helps to inject {@link UserService} related links.
+ *
+ * @author Anatoliy Bazko
+ */
+public final class LinksInjector {
+
+    public static UserDescriptor injectLinks(UserDescriptor userDescriptor, ServiceContext serviceContext) {
+        final User currentUser = EnvironmentContext.getCurrent().getUser();
+        final UriBuilder uriBuilder = serviceContext.getBaseUriBuilder();
```

Comment:
WDYT about using getServiceUriBuilder() method from ServiceContext and remove all .path(UserProfileService.class)?